plopfile.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Defines a plop (https://plopjs.com/) generator to scaffold out new components
+ *
+ * Run with `yarn run plop` or `npm run plop`
+ */
+export default function (plop) {
+  plop.setGenerator('component', {
+    description: 'Add a new component',
+    prompts: [
+      {
+        type: 'input',
+        name: 'name',
+        message: 'component name (e.g. Panel)',
+      },
+      {
+        type: 'input',
+        name: 'group',
+        message: 'component group directory (e.g. layout)',
+      },
+      {
+        type: 'list',
+        name: 'category',
+        message: 'component category',
+        choices: ['presentational', 'composite', 'simple'],
+        default: 'presentational',
+      },
+      {
+        type: 'confirm',
+        name: 'addPatternLibraryPage',
+        message: 'Add a pattern-library documentation page for this component?',
+        default: false,
+      },
+    ],
+    actions: function (data) {
+      const actionList = [
+        {
+          type: 'add',
+          path: 'src/components/{{group}}/{{name}}.js',
+          templateFile: 'plop-templates/{{category}}-component.hbs',
+        },
+        {
+          type: 'add',
+          path: 'src/components/{{group}}/index.js',
+          skipIfExists: true,
+        },
+        {
+          type: 'modify',
+          path: 'src/components/{{group}}/index.js',
+          pattern: /\n\n*$/g,
+          template: `\nexport { default as {{name}} } from './{{name}}';\n`,
```

Comment:
This currently appends the line at the end without sorting the components. It would nice to either sort the imports automatically (eg. via a `transform` helper) or indicate to the user that they should do this before submitting a PR. Same for the tests.

Reply:
I spent about 20 minutes on this (transform helper) before deciding it was more trouble than it was worth at the time. It _is_ mentioned in the "new component checklist" but I'm happy to add another output message similar to the routing one.  I'll give this a proverbial think.